caffe2/python/modeling/initializers.py

```diff
@@ -29,6 +29,33 @@ def create_param(self, param_name, init_net, shape):
             shape=shape,
         )
 
+class pFP16Initializer(object):
+    def __init__(self, operator_name=None, **kwargs):
+        self.operator_name = operator_name
+        self.operator_kwargs = kwargs
+
+    def update(self, operator_name, kwargs):
+        if self.operator_name is not None:
+            raise Exception("Operator name overwrites are not allowed")
+        self.operator_name = operator_name
+        self.operator_kwargs = kwargs
+
+    def create_param(self, param_name, init_net, shape):
```

Comment:
you could call base class method and then add a conversion part + modify ParameterInfo object returned from the base class.